### ISSUE TITLE
Reaktor 6 almost unusable due to resizing problems

### ISSUE THREAD

**Issue by max-phase** [robbert-vdh/yabridge#158]
### Thanks for giving yabridge a shot!

- [X] I read through both the troubleshooting and the known issues sections, and my issue wasn't listed there

### Problem description

Most VSTs are fixed in size, but for the resizeable ones I made the following observations:

All VST2: resizing the window is ignored by the VST. This only adds "gray borders" (system 'panel' color) or crops the plugin.

(I am using Reaper. There are two options for plugin UIs:
'normal' in an "FX" window, consisting of a list of all track plugins on the left and the currently selected plugin's UI on the right
'undocked' (by double-clicking the plugin entry in the FX window): plugin UI has its own window)

Voxengo Span:
VST2 [un]docked: resizing via bottom-right handle (BRH): resize works relatively quick, but the whole FX window moves pixel-wise diagonally towards screen(0, 0).
VST3 [un]docked: resizing the window works flawlessly and updates pretty fast (no BRH available)

iZotope Iris 2:
VST2 [un]docked: resizing via BRH: quite stuttering and needs several updates to resize the FX window (That is the plugin resizes to "bigger than window" and after several slight mouse movements the actual window adjusts).
VST3 [un]docked: resizing via BRH: similar to VST2; resizing via window: window does not move, but plugin takes considerably longer to update (seems to stutter/redraw for every received mouseMoved update), but eventually has its correct size.

Melodyne:
no VST2
VST3 [un]docked: resizing is similar to Iris(VST3), but UI updates quicker (but not fast); does not have BRH

Kontakt 6:
VST2 [un]docked: similar to Voxengo Span, quick resize. Besides the window up-left-move kinda flawless.
VST3 [un]docked: same as VST2 - does not react to window size changes, only to BRH-resizing.

Reaktor 6:
VST2 [un]docked: unusable small after "open/new ensemble", but if one does manage to open an instrument/ensemble UI will update to its minimum size, making the instrument visible. Sometimes does not update UI at all (e.g. clicking on 'EDIT'), then undocking/redocking helps. Resizing the window itself does not add a "panel colored" border, but just increases window size (the empty area shows whatever hangs around in GPU memory).
no VST3

Reaktor 5:
VST2 [un]docked: similar to Reaktor 6, but the initial window does render fully instead of being cropped. No resize handles and does not react to window size changes, therefore cannot be resized.
no VST3

Tests were done using window manager i3. Reaktor shows same behavior using KDE, Xmonad, or WMII instead.
The up-left-moving behavior only occurs on i3, so possibly an i3 bug.

![reaktor6-vst2-after-start](https://user-images.githubusercontent.com/88492091/148607394-2b596cd1-250f-4613-be5d-cfccd47d3c40.jpg)

https://user-images.githubusercontent.com/88492091/148608525-669382fa-3340-4045-b4b3-01a1ef770939.mp4

### What did you expect to happen?

Resizing window does resize VST if resizable.

### What actually happened?

Strange things.

### Operating system

5.15.11-gentoo-x86_64

### How did you install yabridge?

build from master 263b192daa316f80

### yabridge version

263b192daa316f80

### yabridgectl version

263b192daa316f80

### Wine version

wine-staging-6.23

### Plugin

NI Reaktor 6

### Plugin type

VST2

### Plugin architecture

64-bit

### Host

Reaper 6.33

### Desktop environment or WM

i3 & KDE Plasma 5.23.4

**Comment by max-phase** [comment]
I tried setting `editor_xembed`, but that made things worse.

**Comment by robbert-vdh** [comment]
VST2 plugins cannot be resized by resizing host's window. VST3 does have that feature.

**Comment by max-phase** [comment]
If the plugin must be resized by its BRH, then that is ok - but Reaktor does not render any resizing handle, as it is cropped by default (as in the screenshot).
Running it as a standalone app via wine64 does work and renders correctly.

**Comment by robbert-vdh** [comment]
Do you have REAPER's plugin bridge feature enabled? If so, that's still broken on Linux, and it will cause UIs to be cropped to about the size in your screenshot.

**Comment by max-phase** [comment]
I remembered such a setting in the dark corners of my mind..
After reading the Internet twice I found:

The bridging/firewall settings in 'options' are only in the Windows version. Yet, in the FX-add window one can right-click a plugin:
![reaper-bridging-options](https://user-images.githubusercontent.com/88492091/148618051-cafdb523-5bc3-46a2-bcc2-6d9c48fbb0da.jpg)

Default is "separate process". Setting this to "native" for Reaktor does indeed solve the resize and freeze issues!

Thank you so much for solving this!
